### PR TITLE
Feature/mz tab exporter mod support

### PR DESCRIPTION
**Pull request by timosachsenberg**
@hendrikweisser collapsed multiple input parameters into one, minor refactoring and some correction for reporting of terminal modifications.

### CODE REVIEW

File: src/topp/MzTabExporter.cpp [hunk 1]
```diff
@@ -110,12 +110,8 @@ namespace OpenMS
 
     void registerOptionsAndFlags_()
     {
-      registerInputFile_("in_feature", "<file>", "", "FeatureXMLs used to generate the mzTab file.", false);
-      setValidFormats_("in_feature", ListUtils::create<String>("featureXML"));
-      registerInputFile_("in_consensus", "<file>", "", "ConsensusXMLs used to generate the mzTab file.", false);
-      setValidFormats_("in_consensus", ListUtils::create<String>("consensusXML"));
-      registerInputFile_("in_id", "<file>", "", "Identifications used to generate the mzTab file.", false);
-      setValidFormats_("in_id", ListUtils::create<String>("idXML"));
+      registerInputFile_("in", "<file>", "", "FeatureXMLs used to generate the mzTab file.", false);
```
Comment (hendrikweisser) [inline src/topp/MzTabExporter.cpp]: Should say "Input file" instead of "FeatureXMLs".

File: src/topp/MzTabExporter.cpp [hunk 2]
```diff
@@ -568,31 +541,8 @@ namespace OpenMS
...
-              vector<std::pair<Size, MzTabParameter> > pos;
-              pos.push_back(make_pair(ai + 1, MzTabParameter()));
-              mod.setPositionsAndParameters(pos);
-              mods.push_back(mod);
-            }
-          }
-        }
-        mod_list.set(mods);
-        row.modifications = mod_list;
+        // extract all modifications in the current sequence for reporting. In contrast to peptide and protein section all modifications are reported.
```
Comment (hendrikweisser) [inline src/topp/MzTabExporter.cpp]: If you write sentences, it looks better if you capitalize the first word.

File: src/topp/MzTabExporter.cpp [hunk 3]
```diff
@@ -712,7 +662,62 @@ namespace OpenMS
 
       return mztab;
     }
- 
+
+    // generate list of modifications from modified AASequence object. All passed fixed modifications are not reported (PRT and PEP section).
```
Comment (hendrikweisser) [inline src/topp/MzTabExporter.cpp]: "modified AASequence object"? Modified how? Or do you mean potentially storing a sequence with modifications?

File: src/topp/MzTabExporter.cpp [hunk 4]
```diff
@@ -712,7 +662,62 @@ namespace OpenMS
...
+        {
+          if (aas.isModified(ai))
+          {
+            MzTabModification mod;
+            String mod_name = aas[ai].getModification();
+            // MzTab standard is to just report Unimod accession.
+            MzTabString unimod_accession = MzTabString(mod_db->getModification(aas[ai].getOneLetterCode(), mod_name, ResidueModification::ANYWHERE).getUniModAccession());
+            vector<std::pair<Size, MzTabParameter> > pos;
+            pos.push_back(make_pair(ai + 1, MzTabParameter()));
+            mod.setPositionsAndParameters(pos);
```
Comment (hendrikweisser) [inline src/topp/MzTabExporter.cpp]: Swap this line and the next one, to have the same order in all three cases (N-term, anywhere, C-term). Doesn't make a difference now, but might aid in extending or refactoring later.

**Comment by hendrikweisser** [comment]
Works as advertised - no more warnings about modification sites. See my comments on the code; otherwise :+1:.

File: src/topp/MzTabExporter.cpp (hunk 3 above)
Comment (timosachsenberg) [inline src/topp/MzTabExporter.cpp]: AASequence with modifications - will adapt

**Pull request merged**